Integrate gradle versions plugin

omihaz opened the pull request:
Very useful plugin for update dependencies to latest versions.
https://github.com/ben-manes/gradle-versions-plugin

Keep dependencies latest stable - more quality - less bugs.

Current report:
```
------------------------------------------------------------
:app Project Dependency Updates (report to plain text file)
------------------------------------------------------------

The following dependencies are using the latest milestone version:
 - com.android.support:support-annotations:23.1.1
 - com.android.support.test:rules:0.5
 - com.android.support.test:runner:0.5
 - com.facebook.stetho:stetho:1.3.1
 - com.facebook.stetho:stetho-okhttp3:1.3.1
 - com.frogermcs.androiddevmetrics:androiddevmetrics-runtime:0.4
 - com.frogermcs.androiddevmetrics:androiddevmetrics-runtime-noop:0.4
 - com.github.pedrovgs:lynx:1.6
 - com.google.code.findbugs:findbugs:3.0.1
 - com.google.dagger:dagger:2.4
 - com.google.dagger:dagger-compiler:2.4
 - com.jakewharton:butterknife:8.0.1
 - com.jakewharton:butterknife-compiler:8.0.1
 - com.jakewharton:process-phoenix:1.0.2
 - com.jakewharton.picasso:picasso2-okhttp3-downloader:1.0.2
 - com.jakewharton.timber:timber:4.1.2
 - com.squareup.okhttp3:logging-interceptor:3.2.0
 - com.squareup.okhttp3:mockwebserver:3.2.0
 - com.squareup.okhttp3:okhttp:3.2.0
 - com.squareup.picasso:picasso:2.5.2
 - hu.supercluster:paperwork:1.2.7
 - io.reactivex:rxjava:1.1.5
 - junit:junit:4.12
 - net.sourceforge.pmd:pmd-java:5.4.1
 - nl.littlerobots.rxlint:rxlint:1.0
 - org.jacoco:org.jacoco.agent:0.7.6.201602180812

The following dependencies exceed the version found at the milestone revision level:
 - com.squareup.leakcanary:leakcanary-android [1.4-beta2 <- 1.3.1]

The following dependencies have later milestone versions:
 - com.android.support:appcompat-v7 [23.1.1 -> 24.0.0-alpha2]
 - com.android.support:cardview-v7 [23.1.1 -> 24.0.0-alpha2]
 - com.android.support:design [23.1.1 -> 24.0.0-alpha2]
 - com.android.support:recyclerview-v7 [23.1.1 -> 24.0.0-alpha2]
 - com.android.support:support-v4 [23.1.1 -> 24.0.0-alpha2]
 - com.android.support.test.espresso:espresso-contrib [2.2.1 -> 2.2.2]
 - com.android.support.test.espresso:espresso-core [2.2.1 -> 2.2.2]
 - com.fasterxml.jackson.core:jackson-databind [2.6.3 -> 2.7.4]
 - com.github.brianPlummer:tinydancer [0.0.8 -> 0.0.9]
 - com.google.auto.value:auto-value [1.2-rc1 -> 1.2]
 - com.puppycrawl.tools:checkstyle [6.16.1 -> 6.18]
 - com.squareup.retrofit2:adapter-rxjava [2.0.0-beta4 -> 2.0.2]
 - com.squareup.retrofit2:converter-jackson [2.0.0-beta4 -> 2.0.2]
 - com.squareup.retrofit2:retrofit [2.0.0-beta4 -> 2.0.2]
 - com.yandex.android:mobmetricalib [2.40 -> 2.41]
 - net.orfjackal.retrolambda:retrolambda [2.1.0 -> 2.3.0]
 - nl.jqno.equalsverifier:equalsverifier [1.7.5 -> 2.0.2]
 - org.aspectj:aspectjrt [1.8.8 -> 1.8.9]
 - org.assertj:assertj-core [2.4.0 -> 3.4.1]
 - org.jacoco:org.jacoco.agent [0.7.1.201405082137 -> 0.7.6.201602180812]
 - org.mockito:mockito-core [1.10.19 -> 2.0.52-beta]
 - org.robolectric:robolectric [3.0 -> 3.1-rc1]
```

vanniktech commented:
This would fix #160

artem-zinnatullin commented on build.gradle:
```diff
@@ -71,6 +74,10 @@ subprojects {
         checkTask.dependsOn('checkstyle')
 
+        def buildTask = tasks.findByName('build')
+        buildTask.dependsOn('dependencyUpdates')
```
Can we don't do this? This task is so slow for me (it's useful though) that it's not worth it to run it on every build

artem-zinnatullin commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.12-all.zip
```
![fry_looking_squint](https://cloud.githubusercontent.com/assets/967132/15935119/4414bcc6-2e77-11e6-86fd-3674391954e8.jpg)

omihaz commented:
Hello.

About gradle wrapper, Android Studio suggests to use gradle with source (-all build).

I updated PR, removed gradle wrapper changes and removed dependencyUpdates task from build.

artem-zinnatullin commented:
Yeah, I'm just joking :)

artem-zinnatullin commented:
I'm ok with merging that, please rebase on latest repo state

omihaz commented:
Rebased.

vanniktech commented:
👍

vanniktech commented on dependencies.gradle:
```diff
@@ -17,6 +17,7 @@ ext.versions = [
         errorProneVersion            : '0.0.8',
         errorProneCoreVersion        : '2.0.9',
         dexcountPlugin               : '0.5.4',
+        versionsGradlePlugin         : '0.12.0',
```
There's actually a new version (0.13.0)

omihaz commented:
Updated plugin to latest version. PR from May.

lenguyenthanh commented:
👍

artem-zinnatullin commented:
Thanks, @omihaz!
